src/components/entry/Entry.jsx

```diff
@@ -1,14 +1,24 @@
 import { StyledEntry } from './Entry.styled';
 import Typography from '../ui/Typography';
 
-export default function Entry({ entry, location }) {
+export default function Entry({ entry, locations }) {
 	return (
 		<StyledEntry>
 			<Typography variant="h3">Name:</Typography>
 			<Typography variant="body1">{entry.name}</Typography>
-			<Typography>Place</Typography>
-			<Typography variant="body1">long: {location.location[0]}</Typography>
-			<Typography variant="body1">lat: {location.location[1]}</Typography>
+			<Typography variant="h3">Place:</Typography>
+			<Typography variant="body1">
+				long:
+				{locations.map(location => {
+					return location.entryId === entry.id && location.location[0];
+				})}
```

Comment:
Hey Martin :) We decided in class contrary, because this enhances the readability. 

Cases we separate logic from jsx usually are: 
- mulitple use 
- "a lot" of function-calls

Reply:
Thank you for the insightful words. But still not sure how this:
```js
<Typography variant="body1">
    long:
    {locations.map(location => {
	return location.entryId === entry.id && location.location[0];
    })}
</Typography>
```
 enhances the readability over this:
```js
<Typography variant="body1">
    long: {long}
</Typography>
```
 🤷‍♂️

Reply:
`long: {long}` unfortunately reveals nothing about the logic that is applied there, so scrolling and a lot of eye movement are often necessary to see which logic is used in the jsx.

I totally understand how this extracted logic makes the jsx more readable but it harms the readability of the component as a whole, which seems to be avoided. :)

Reply:
Thank you very much for your detailed feedback. this helped me a lot not to make such gross mistakes at the second pr . I created a helper function and let create the respective location with find() now. Everything works, but I think I will implement Dominik's tip now after all and rebuild my data structure. kiss is such a wonderful motto.